src/core/Directus/Services/TablesService.php

```diff
@@ -735,6 +743,29 @@ public function getTableObject($tableName)
         return SchemaService::getCollection($tableName);
     }
 
+    public function hasPrimaryField($fields){
+        $result = false;
+
+        foreach($fields as $field){
+            if($field['primary_key'] === true){
+                $result = true;
+                break;
+            }
+        }
+        return $result;
+    }
+
+    public function hasUniquePrimaryKey($fields){
+        $primaryKeyCount = 0;
+        foreach($fields as $field){
+            if($field['primary_key'] === true){
+                $primaryKeyCount++;
+            }
+        }
+        return $primaryKeyCount <= 1;
```

Comment:
Hey @danielfrentz, this must be `$primaryKeyCount === 1` as this will fail when there's not primary key by returning `true`. When checking the condition `0 <= 1`.

Reply:
This may be too small of a point to argue about but heres my case for <= over ===.

Theres 2 separate checks. The first checks to see if the key exists at all, while this one checks that there is a maximum of 1. 
If it were changed to === then from a maintainability point of view, it would make life a little harder since if you remove the other check, or move it so that it happens afterwards,  then this one will trigger when theres only 1 primary key and give a bad error.

Reply:
The intention of this is to tell whether there's only or just if there's not more than one?

These are two public function that does a separated validation and it shouldn't depend on a other function result indirectly, like whether or not the first failed or not.

Please add comments to the functions definition to make sure what really does, and it should be good!

Reply:
Hey @danielfrentz – any update on this? We'd love to get it merged/closed! Thanks again!